src/views/Graph.vue

```diff
@@ -391,6 +647,268 @@ export default {
   },
 
   methods: {
+    /**
+     * A nested Family object
+     * @typedef {Object} Family
+     * @property {Family[]} children - Array of the familys child families
+     * @property {number} id - An integer id (required for vuetify toolbar)
+     * @property {string} name - The family name
+     * @property {boolean} disabled - used in vuetify toolbar
+     */
+    /**
+     * A node object
+     * @typedef {Object} Node
+     * @param {Array} children - The nodes immediate children
+     * @param {Undefined} familyTree - The nodes familyTree
+     * @param {String} id - The node id
+     * @param {String} name - The node name
+     * @param {Object} node - The node object (note not of type Node)
+     * @param {String} parent - The nodes immediate parent id
+     * @param {Object} tokens - The nodes token object
+     * @param {String} type - The nodes type "task" | "$namespace" | "$edge"
+     */
+    /**
+     * Get a nested object of families
+     *
+     * @returns {Family[]} array containing nested structure of families
+     */
+    getTree () {
+      const counter = {
+        value: 0,
+        next () {
+          this.value = this.value + 1
+          return this.value
+        }
+      }
+
+      const root = {
+        id: counter.next(),
+        name: 'root',
+        children: []
+      }
+      if (this.workflows) {
+        const tokens = this.workflows[0].tokens.clone({ cycle: '$namespace|root' })
+        const node = this.cylcTree.$index[tokens.id]
+        if (node) {
+          return this.getTreeHelper(root, node, counter).children
+        }
+      }
+    },
+    /**
+     * Get a nested object of families
+     * @property {Family} store - nested object of families
+     * @property {Node} node - node object
+     * @property {number} counter - counter used for index
+     * @returns {Family} nested structure of families
+     */
+    getTreeHelper (store, node, counter) {
+      let tempItem
+      const isParent = this.collapseFamily.includes(node.name)
+      const isAncestor = this.allParentLookUp[node.name].some(element => {
+        return this.collapseFamily.includes(element)
+      })
+      const disabled = isParent || isAncestor
+      for (const childFamily of node.node.descendants) {
+        if (this.namespaces.map((obj) => obj.name).includes(childFamily)) {
```

Comment:
Can short-circuit this
```suggestion
        if (this.namespaces.some((obj) => obj.name === childFamily)) {
```